src/main/java/org/wso2/scim2/operation/GroupOperations.java

```diff
@@ -150,7 +159,16 @@ public void updateGroup() throws IdentitySCIMException {
                 if (groupId == null) {
                     return;
                 }
-                String encodedGroup = scimClient.encodeSCIMObject((AbstractSCIMObject) scimObject, SCIMConstants.JSON);
+
+                String encodedGroup;
+                List<String> users = ((Group) scimObject).getMembersWithDisplayName();
+                if (CollectionUtils.isEmpty(users)) {
+                    encodedGroup = scimClient.encodeSCIMObject((AbstractSCIMObject) scimObject, SCIMConstants.JSON);
+                } else {
+                    // Find corresponding userIds of group members and enrich the scimObject.
+                    Group updatedGroup = addUserIDForMembersOfGroup();
```

Comment:
The SCIM2 group API expects members in this format.
```
{
    "display": "smith",
    "value": "6b39f31d-9474-48b0-a862-e181d9e09810"
}
```

Hence user-id is resolved and passed. 
In previous versions passing username was enough, but in latest implementations user-id need to be passed.